### DISCUSSION TITLE
Handling JSON references using $ref

### DISCUSSION THREAD

**Discussion by bbDSA** [nlohmann/json#3093] [Q&A] [answered]
Hello,

will the library automatically resolve reference which are using $ref (possibly after providing a URI resolver). If so, would the reference then be fully resolved to a plain JSON object and would that JSON object share the memory with other occurrences of this $ref? Can I disable the processing of $ref?

Why am I asking? I have an OpenAPI schema which makes heavy use of $ref an references a vast number of schema several times. If the referenced schema would be resolved to a JSON object which allocates its own memory, space consumption could become a real issue. Or would I have to implement the $ref handling anyways on my own to use shared memory when necessary?

Best regards,
Boris

**Comment by nlohmann** [comment, marked as answer]
No, since `$ref` is part of JSON Schema if I understand correctly. You may want to check out https://github.com/pboettch/json-schema-validator which may help you with this.
